src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,66 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Arrays;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_SIZE = 10;
+    private static final int KEY_NOT_EXIST_MARKER_INDEX = -1;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        this.keys = (K[]) new Object[0];
+        this.values = (V[]) new Object[0];
```

Comment:
`this` keyword is redundant here
> If you use arrays for your solution, we assume that the maximum number of elements in our Storage is 10.

if so, remove any `resize` logic
```suggestion
        keys = (K[]) new Object[MAX_STORAGE_SIZE];
        values = (V[]) new Object[MAX_STORAGE_SIZE];
```